include/myst/thread.h

```diff
@@ -342,50 +344,81 @@ MYST_INLINE bool myst_is_zombied_process(myst_process_t* process)
 
 typedef struct myst_thread_queue
 {
-    myst_thread_t* front;
-    myst_thread_t* back;
+    myst_thread_t* _front;
+    myst_thread_t* _back;
 } myst_thread_queue_t;
 
 MYST_INLINE size_t myst_thread_queue_size(myst_thread_queue_t* queue)
 {
     size_t n = 0;
 
-    for (myst_thread_t* p = queue->front; p; p = p->qnext)
+    for (myst_thread_t* p = queue->_front; p; p = p->_qnext)
         n++;
 
     return n;
 }
 
-MYST_INLINE void myst_thread_queue_push_back(
+MYST_INLINE void myst_thread_queue_push_back_bitset(
     myst_thread_queue_t* queue,
-    myst_thread_t* thread)
+    myst_thread_t* thread,
+    uint32_t bitset)
 {
-    thread->qnext = NULL;
+    thread->_qnext = NULL;
+    thread->_bitset = bitset;
 
-    if (queue->back)
-        queue->back->qnext = thread;
+    if (queue->_back)
+        queue->_back->_qnext = thread;
     else
-        queue->front = thread;
+        queue->_front = thread;
 
-    queue->back = thread;
+    queue->_back = thread;
 }
 
-MYST_INLINE myst_thread_t* myst_thread_queue_pop_front(
+MYST_INLINE void myst_thread_queue_push_back(
+    myst_thread_queue_t* queue,
+    myst_thread_t* thread)
+{
+    myst_thread_queue_push_back_bitset(queue, thread, FUTEX_BITSET_MATCH_ANY);
+}
+
+MYST_INLINE myst_thread_t* myst_thread_queue_get_front(
     myst_thread_queue_t* queue)
 {
-    myst_thread_t* thread = queue->front;
+    if (queue)
+        return queue->_front;
+
+    return NULL;
+}
+
+MYST_INLINE myst_thread_t* myst_thread_queue_pop_front_bitset(
```

Comment:
The bitset seems unrelated to the thread queue operation (bitset is a member of the thread not the queue).

Reply:
the naming of this operation signifies the ability to get the bitset data when popping the thread queue as well. Do you have suggestions on how to handle this?